src/static_array.cr

```diff
@@ -67,31 +111,74 @@ struct StaticArray(T, N)
     indexes.map { |index| self[index] }
   end
 
+  # Calls the given block at the given index.
+  # Raises `IndexError` if trying to set an element outside the array's range.
+  #
+  # ```
+  # array = StaticArray(Int32, 3).new { |i| i+1 } # => [1, 2, 3]
+  # array.update(1) { |x| x * 2 } # => [1, 4, 3]
+  # array.update(5) { |x| x * 2 } # => IndexError
+  # ```
   def update(index : Int)
     index = check_index_out_of_bounds index
     to_unsafe[index] = yield to_unsafe[index]
   end
 
+  # Returns the size of `self`
+  #
+  # ```
+  # array = StaticArray(Int32, 3).new { |i| i+1 }
+  # array.size # => 3
+  # ```
   def size
     N
   end
 
+  # Returns all the elements of the array by the given value
+  # 
+  # ```
+  # array = StaticArray(Int32, 3).new { |i| i+1 }
+  # array.[]= 2 # => [2, 2, 2]
```

Comment:
It's valid Ruby too :-)

But yes, it can be a bit cryptic. We better change it to fill, so it's similar to Array.